code/src/java/pcgen/gui2/PCGenActionMap.java

```diff
@@ -248,11 +251,12 @@ private DebugAction()
 		@Override
 		public void actionPerformed(ActionEvent e)
 		{
+			GuiAssertions.assertIsNotJavaFXThread();
 			if (dialog == null)
 			{
-				dialog = new DebugDialog();
+				Platform.runLater(() -> dialog = new DebugDialog());
```

Comment:
Why not all the if login on the same thread call? If null it ends up using two different threads.

Reply:
good call. I'll fix this